src/plugins/chart_expressions/expression_xy/public/components/xy_chart.tsx

```diff
@@ -330,13 +304,9 @@ export function XYChart({
     const fit = !hasBarOrArea && extent.mode === 'dataBounds';
     let min: number = NaN;
     let max: number = NaN;
-
     if (extent.mode === 'custom') {
-      const { inclusiveZeroError, boundaryError } = validateExtent(hasBarOrArea, extent);
-      if (!inclusiveZeroError && !boundaryError) {
-        min = extent.lowerBound ?? NaN;
-        max = extent.upperBound ?? NaN;
-      }
+      min = extent.lowerBound ?? NaN;
```

Comment:
These validations are necessary - otherwise it's possible to go to line chart, configure custom extends which do not include zero, then switch back to bar chart.

Reply:
I can say even more. I've faced a terrifying situation today, which has changed my mind related to the validations I was adding to the expressions. All the configurations added to the panel in the Lens will be spread to all the suggestions. So, there can be cases, where the errors will be thrown instead of drawing the chart. Before, the behavior was overriding inside the expressions and renderers. For now, it is moved to the `toExpression` and `toPreviewExpression` functions of Lens for making the logic more obvious.

What about `NaN` at `min`/`max`, I've moved the validation to the `xyVis` and overridden `min`/`max` values with `NaN` for `layeredXyVis` in case of invalid input.